packages/block/src/header.ts

```diff
@@ -491,9 +489,12 @@ export class BlockHeader {
     return this._common.hardfork() || this._common.activeHardfork(this.number.toNumber())
   }
 
-  private async _getBlockByHash(blockchain: Blockchain, hash: Buffer): Promise<Block | undefined> {
+  private async _getHeaderByHash(blockchain: Blockchain, hash: Buffer): Promise<BlockHeader | undefined> {
     try {
-      return blockchain.getBlock(hash)
+      return new Promise(async (resolve) => {
+        let header = (await blockchain.getBlock(hash)).header 
+        resolve(header)
+      })
```

Comment:
haven't tried this in the branch yet but maybe something like
```suggestion
      const block = await blockchain.getBlock(hash)
      return block ? block.header : undefined
```

Reply:
Yeah I will add something like this, ESLint doesn't like the current implementation